### PR TITLE
[WIP] Make 64bit DT work with 32bit DF on macOS

### PR DESCRIPTION
**Pull request by ifreund**
This apparently only fixed part of #38 because DT still crashes when used with 32bit DF.  However, this code does properly set m_pointer_size based on the architecture of the executable.

### CODE REVIEW

File: src/dfinstanceosx.mm [hunk 1]
```diff
@@ -205,7 +206,22 @@ of this software and associated documentation files (the "Software"), to deal
...
     m_alloc_end = 0;
-    m_pointer_size = sizeof(VIRTADDR); // TODO: set pointer size
+    
+    std::ifstream dfexe(m_loc_of_dfexe.toUtf8().constData(), std::ios::in | std::ios::binary);
+    if(!dfexe.is_open()) 
+        LOGE << "Failed to open dwarfort.exe (" << m_loc_of_dfexe << ")";
+    else {
+        uint32_t magic_number;
+        dfexe.read(reinterpret_cast<char*>(&magic_number), sizeof(magic_number));
+        if(magic_number == 0xFEEDFACE) // 32bit
```
Comment (cvuchener) [inline src/dfinstanceosx.mm]: Use MH_MAGIC

File: src/dfinstanceosx.mm [hunk 2]
```diff
@@ -205,7 +206,22 @@ of this software and associated documentation files (the "Software"), to deal
...
+            m_pointer_size = 4;
+        else if(magic_number == 0xFEEDFACF) // 64bit
```
Comment (cvuchener) [inline src/dfinstanceosx.mm]: Use MH_MAGIC_64

**Comment by cvuchener** [comment]
When does it crash exactly? I don't know what could be wrong when reading. But writing strings may need to be changed, did you check it?

By the way, most the string related stuff is in DFInstanceNix. It used to be inherited by DFInstanceLinux, but, now, it is only used by DFInstanceOSX. You can merge the two classes if you want.

**Comment by ifreund** [comment]
Seems to crash on 
https://github.com/Dwarf-Therapist/Dwarf-Therapist/blob/ce4ec74ded956190354fa2bb03867a3c1686ade8/src/dfinstance.cpp#L1087

I've been super busy recently, but I'll probably have some more time to debug tonight.

>By the way, most the string related stuff is in DFInstanceNix. It used to be inherited by DFInstanceLinux, but, now, it is only used by DFInstanceOSX. You can merge the two classes if you want.

I'll get to this at some point after getting everything working.

**Comment by ifreund** [comment]
Closing this as I unfortunately never got around to finishing what I started and no longer have the mac I was using to test this on.

**Pull request closed without merging**